process/coordinator/processProposal.go

```diff
@@ -123,23 +144,28 @@ func (tc *transactionCoordinator) CreateMbsCrossShardDstMe(
 
 // SelectOutgoingTransactions returns transactions originating in the shard, for a block proposal
 func (tc *transactionCoordinator) SelectOutgoingTransactions() [][]byte {
-	txHashes := make([][]byte, 0)
-	var currentTxHashes [][]byte
-	var err error
+	selectedTxHashes := make([][]byte, 0)
+	selectedTxs := make([]data.TransactionHandler, 0)
 	for _, blockType := range tc.preProcProposal.keysTxPreProcs {
 		txPreProc := tc.preProcProposal.getPreProcessor(blockType)
 		if check.IfNil(txPreProc) {
 			log.Warn("transactionCoordinator.SelectOutgoingTransactions: getPreProcessor returned nil for block type", "blockType", blockType)
 			continue
 		}
 
-		currentTxHashes, err = txPreProc.SelectOutgoingTransactions()
+		txHashes, txs, err := txPreProc.SelectOutgoingTransactions()
```

Comment:
I think we could check the remaining bandwidth here and pass it to the selectOutgoingTransactions.
Then we wouldn't need to add the gas computation component on preprocessors.

Reply:
indeed, applied the suggestion